src/app.js

```diff
@@ -1,133 +1,100 @@
-if (config.SENTRY_DSN) {
-  Raven.config(config.SENTRY_DSN)
-    .install();
+'use strict'
+
+// get configuration
+const {
+    SENTRY_DSN,
+    MAX_POST_SIZE,
+    FALLBACK,
+    RABBIT
+} = require('./config')
+
+const express = require('express')
+const morgan = require('morgan')
+const bodyParser = require('body-parser')
+const Raven = require('raven')
+
+//TODO Upgrade to sentry pls -- this is deprecated
+if (SENTRY_DSN) {
+    Raven.config(SENTRY_DSN)
+        .install()
 }
 
-if (config.FALLBACK) {
-  // Use var intentionally so that rabbit doesn't need to be defined
-  var rabbit = require('./rabbit'); // eslint-disable-line no-var
-}
+const prom = require('./lib/prometheus')
+const logger = require('./lib/logger')
+const queue = require('./lib/queue')
+const ServiceError = require('./lib/ServiceError')
+const rabbitController = require('./lib/rabbit')
 
-let prom = require('./prometheus');
-
-let logger = require('./logger');
-let pubsub = require('./pubsub');
-let queue = require('./queue');
-
-// bootstrap app
-let app = express();
-app.use(morgan('dev'));
-app.use(bodyParser.json({ limit: config.MAX_POST_SIZE }));
-app.use(bodyParser.urlencoded({extended: false}));
-
-// register routes
-app.get('/', (req, res, next) => {
-  res.json({ping: 'pong'});
-});
-
-app.get('/healthz', (req, res, next) => {
-  res.json({ping: 'pong'});
-});
-
-app.post('/messages/:channel', (req, res, next) => {
-  let end = prom.requestSummary.startTimer();
-  let channel = req.params.channel;
-  prom.receiveCount.inc({channel});
-  let messages = req.body.messages || [];
-  queue.push(publishJob(channel, messages, end));
-  res.json({success: true});
-});
-
-
-  /**
-   * Return a job
-   *
-   * @param {String} channel
-   * @param {[Object]} messages
-   * @param {Function} end
-   * @param {Integer} retries
-   *
-   * @return {Function}
-   */
-function publishJob(channel, messages, end, retries=0) {
-  return function(cb) {
-    pubsub.publish(channel, messages).then((result)=>{
-      let errors = result.errors;
-      if(errors.length > 0) {
-        prom.publishCount.inc({state: 'failed', channel});
-        if(config.FALLBACK && retries >= 2) {
-          end();
-          return rabbit.publish(channel, errors);
-        }
-        retries++;
-        queue.push(publishJob(channel, errors.map((error) => error.message), end, retries));
-      } else {
-        prom.publishCount.inc({state: 'success', channel});
-        end();
-      }
-      cb();
-    }).catch((error) => {
-      cb(error);
-    });
-  };
-}
+if (FALLBACK) rabbitController.start()
+
+const app = express()
+
+app.use(morgan('dev'))
+app.use(bodyParser.json({ limit: MAX_POST_SIZE }))
+app.use(bodyParser.urlencoded({ extended: false }))
+
+// why is this here? :shrug:
+app.get('/', async (req, res) => {
+    // logger.info('ping')
+    res.json({ ping: 'pong' }).end()
+})
+
+// k8s health check endpoint
+app.get('/healthz', (req, res) => res.json({ ping: 'pong' }) )
```

Comment:
This appears to still always return a 200. This should return a non 200 when receiving a `SIGTERM`.

Reply:
The standard way to close server connections look like this:

Reply:
There is a flag now.

Reply:
Awesome. Thank you